registration-processor/core-processor/registration-processor-abis-handler-stage/src/test/java/io/mosip/registration/processor/abis/handler/stage/test/AbisHandlerStageTest.java

```diff
@@ -1,500 +1,738 @@
+
+		RegBioRefDto bioRefDto = new RegBioRefDto();
+		bioRefDtos.add(bioRefDto);
+		Mockito.when(packetInfoManager.getBioRefIdByRegId(any())).thenReturn(bioRefDtos);
+
+		Mockito.doNothing().when(packetInfoManager).saveBioRef(any(), any(), any());
+
+		AbisRequestDto abisRequestDto = new AbisRequestDto();
+		abisRequestDto.setAbisAppCode("ABIS1");
+		abisRequestDto.setStatusCode("IN-PROGRESS");
+		abisRequestDtoList.add(abisRequestDto);
+		Mockito.when(packetInfoManager.getAbisRequestsByBioRefId(any())).thenReturn(abisRequestDtoList);
+
+		Mockito.doNothing().when(packetInfoManager).saveAbisRequest(any(), any(), any());
+
+		Mockito.when(packetInfoManager.getDemoListByTransactionId(any())).thenReturn(regDemoDedupeListDtoList);
+
+		PowerMockito.mockStatic(JsonUtils.class);
+		PowerMockito.when(JsonUtils.javaObjectToJsonString(any(AbisInsertRequestDto.class)))
+				.thenReturn("AbisInsertRequestDto");
+		PowerMockito.when(JsonUtils.javaObjectToJsonString(ArgumentMatchers.any(AbisIdentifyRequestDto.class)))
+				.thenThrow(JsonProcessingException.class);
+
+		MessageDTO dto = new MessageDTO();
+		dto.setRid("10003100030001520190422074511");
+		MessageDTO result = abisHandlerStage.process(dto);
+
+		assertTrue(result.getInternalError());
+	}
+
+	@Test
+	public void bioRecordDataShareMisMatch() throws ApisResourceAccessException {
+
+		defaultMockToProcess();
+
+		mockDataSharePolicy(Lists.newArrayList(BiometricType.IRIS));
+
+		MessageDTO dto = new MessageDTO();
+		dto.setRid("10003100030001520190422074511");
+		MessageDTO result = abisHandlerStage.process(dto);
+
+		assertFalse(result.getIsValid());
+	}
+
+	@Test
+	public void bioRecordDataNotFound()
+			throws ApisResourceAccessException, PacketManagerException, JsonProcessingException, IOException {
+
+		defaultMockToProcess();
+
+		Mockito.when(packetManagerService.getBiometrics(any(), any(), any(), any(), any())).thenReturn(null);
+		mockDataSharePolicy(Lists.newArrayList(BiometricType.IRIS, BiometricType.FINGER, BiometricType.FACE));
+
+		MessageDTO dto = new MessageDTO();
+		dto.setRid("10003100030001520190422074511");
+		MessageDTO result = abisHandlerStage.process(dto);
+
+		assertFalse(result.getIsValid());
+	}
+
+	@Test
+	public void biometricsNotFoundWithSegmentConfig()
+			throws ApisResourceAccessException, PacketManagerException, JsonProcessingException, IOException {
+
+		defaultMockToProcess();
+
+//		Mockito.when(packetManagerService.getBiometrics(any(),any(),any(), any(),any())).thenReturn(null);
+		mockDataSharePolicy(Lists.newArrayList(BiometricType.FACE, BiometricType.IRIS, BiometricType.FINGER));
+
+		setMetaInfoMap(Arrays.asList("leftEye"));
+//		setMetaInfoMap(Arrays.asList("rightMiddle"));
+		MessageDTO dto = new MessageDTO();
+		dto.setRid("10003100030001520190422074511");
+		MessageDTO result = abisHandlerStage.process(dto);
+
+		assertFalse(result.getIsValid());
+	}
+	
+	@Test
+	public void biometricsTypeNotFoundConfig()

```

Comment:
Add 1 more test case to have policy for finger and iris and all are marked exception as per meta info

Reply:
added in bioRecordDataNotFound